microsoft/testsuites/mshv/mshv_root_tests.py

```diff
@@ -77,6 +90,155 @@ def verify_mshvlog_is_active(
+        remote_cp.copy_to_remote(
+            src=PurePath(chkbinpath),
+            dest=PurePath(chk_bin_remote_dir),
+        )
+        remote_cp.copy_to_remote(
+            src=PurePath(chkloaderpath),
+            dest=PurePath(chk_loader_remote_dir),
+        )
+
+        tar = node.tools[Tar]
+        tar.extract(
+            file=f"{chk_bin_remote_dir}/{chk_bin_tar_file}",
+            dest_dir=chk_bin_extract_dir,
+            gzip=True,
+            sudo=True,
+        )
+        tar.extract(
+            file=f"{chk_loader_remote_dir}/{chk_loader_tar_file}",
+            dest_dir=chk_loader_extract_dir,
+            gzip=True,
+            sudo=True,
+        )
+
+        # Copy CHK bins into test machine
+        copy_tool = node.tools[Cp]
+        copy_tool.copy(
+            src=PurePath(chk_bin_extract_dir) / "Windows" / "System32",
+            dest=PurePath("/boot/efi/Windows"),
+            sudo=True,
+            recur=True,
+        )
+        copy_tool.copy(
+            src=PurePath(chk_loader_extract_dir) / "boot" / "efi" / "lxhvloader.dll",
+            dest=PurePath("/boot/efi"),
+            sudo=True,
+        )
+
+        # Remove kernel lockdown from grub config
+        node.tools[Sed].substitute(
+            regexp="lockdown=integrity",
+            replacement="",
+            file=grub_config_file,
+            sudo=True,
+        )
+
+        # Add MSHV debug option in chainloader
+        # This is to load hv binaries with debug mode enabled
+        hv_debug_option = "LXHVLOADER_DEBUG=TRUE"
+        grub_config = node.tools[Cat].read(
+            file=grub_config_file,
+            force_run=True,
+            sudo=True,
+        )
+        regex = re.compile(r"(?P<chainloader_cfg>.*chainloader.*)")
+        chainloader_grub_data = find_group_in_lines(
+            lines=grub_config,
+            pattern=regex,
+            single_line=False,
+        )
+        chainloader_config = chainloader_grub_data.get("chainloader_cfg", "").strip()
+        err_msg = f"Cannot get chainloader config, got {chainloader_config}"
+        assert chainloader_config, err_msg
+        if hv_debug_option not in chainloader_config:
+            node.tools[Sed].substitute(
+                regexp="MSHV_SEV_SNP=TRUE",
+                replacement=f"MSHV_SEV_SNP=TRUE {hv_debug_option}",
+                file=grub_config_file,
+                sudo=True,
+            )
+
+        node.tools[Reboot].reboot_and_check_panic(log_path)
+
+        # Trigger HV crash and check if dump is generated
+        hvdbg = node.tools[Ls].path_exists(mshv_debug_sysfs, sudo=True)
+        if not hvdbg:
+            raise LisaException(f"sysfs entry not present: {mshv_debug_sysfs}")
+
+        kdump_util = KdumpUtils(crashdump_timeout=800, kdump_cmd=mshv_crash_command)
+        kdump_util._kdump_test(
+            node=node,
+            log_path=log_path,
+            log=log,
+            is_auto=False,
+        )
+
+        node.mark_dirty()
```

Comment:
It might not work if placed here. Perhaps using a try-catch-finally block would help?

Reply:
Done. Please check